From 8cd592dfcb6a5c41ae0dae0b4ec7571fdfa36f92 Mon Sep 17 00:00:00 2001
From: Will Woods <wwoods@redhat.com>
Date: Wed, 14 Mar 2012 17:09:40 -0400
Subject: [PATCH] fix "execvp: No such file or directory" in emergency_shell

Older versions of setsid emit this message on stderr if you try to run
"setsid --help". Redirect it to /dev/null.
---
 modules.d/99base/dracut-lib.sh |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/99base/dracut-lib.sh b/modules.d/99base/dracut-lib.sh
index a6c4e24..f0d426c 100755
--- a/modules.d/99base/dracut-lib.sh
+++ b/modules.d/99base/dracut-lib.sh
@@ -814,7 +814,7 @@ emergency_shell()
             _ctty=/dev/$_ctty
         fi
         [ -c "$_ctty" ] || _ctty=/dev/tty1
-        strstr "$(setsid --help)" "control" && CTTY="-c"
+        strstr "$(setsid --help 2>/dev/null)" "control" && CTTY="-c"
         setsid $CTTY /bin/sh -i -l 0<$_ctty 1>$_ctty 2>&1
     else
         warn "Boot has failed. To debug this issue add \"rdshell\" to the kernel command line."
